commands/slack.go

```diff
@@ -1,46 +1,38 @@
 package commands
 
 import (
-	"bytes"
-	"fmt"
-	"net/http"
-	"os"
+  "fmt"
+  "github.com/nlopes/slack"
+  "os"
 )
 
-// these are just wrappers for sendMsg
 func startMsg(slackConfig map[string]string, env string) {
-	user := os.Getenv("USER")
-	name := slackConfig["appname"]
-	msg := fmt.Sprintf(":rocket: %s has *started* deploying %s to *%s*!", user, name, env)
+  user := os.Getenv("USER")
+  name := slackConfig["appname"]
+	msg := fmt.Sprintf(":%s: %s has *started* deploying %s to *%s*!", slackConfig["start_emoji"], user, name, env)
 
-	sendMsg(msg, "good", slackConfig)
+	sendMsg(msg, slackConfig)
 }
 
 func finishMsg(slackConfig map[string]string, env string) {
-	user := os.Getenv("USER")
-	name := slackConfig["appname"]
-	msg := fmt.Sprintf(":star2: %s has *finished* deploying %s to *%s*!", user, name, env)
+  user := os.Getenv("USER")
+  name := slackConfig["appname"]
+	msg := fmt.Sprintf(":%s: %s has *finished* deploying %s to *%s*!", slackConfig["finish_emoji"], user, name, env)
 
-	sendMsg(msg, "good", slackConfig)
+	sendMsg(msg, slackConfig)
 }
 
 func errorMsg(slackConfig map[string]string) {
-	msg := ":crocodile: Something went *wrong*!"
+	msg := fmt.Sprintf(":%s: Something went *wrong*!", slackConfig["error_emoji"])
 
-	sendMsg(msg, "danger", slackConfig)
+	sendMsg(msg, slackConfig)
 }
 
-// post request to slack
-func sendMsg(msg string, msgType string, slackConfig map[string]string) {
-	data := []byte(fmt.Sprintf(`{"attachments": [{"mrkdwn_in": ["text"], "color": "%s", "text": "%s"}]}`, msgType, msg))
-	url := slackConfig["webhook"]
+func sendMsg(msg string, slackConfig map[string]string) {
+  api := slack.New(slackConfig["token"])
 
-	r, _ := http.NewRequest("POST", url, bytes.NewBuffer(data))
-	r.Header.Add("Content-Type", "application/json")
-	client := &http.Client{}
-	resp, err := client.Do(r)
+  _, _, err := api.PostMessage(slackConfig["channel"], msg, slack.PostMessageParameters{})
```

Comment:
@onshi yup, using this whole big lib just to make it couple of lines shorter seems like a bit of overkill :( But maybe with more customization options we could delegate payload creating to another function and make it more readable?